be/src/http/http_channel.cpp

```diff
@@ -60,6 +70,10 @@ void HttpChannel::send_reply(HttpRequest* request, HttpStatus status, std::strin
 #endif
     auto evb = evbuffer_new();
     evbuffer_add(evb, content.data(), content.size());
+    if (content_type.has_value()) {
```

Comment:
shall move fix to before line 65, so in case it is a UNIT TEST, the unit test can get the hijacked http request will the header of content-type set properly.

Reply:
line 65-line70 is a hack code for unit test, s_injected_send_reply will be defined in unit test so it can directly get the response body and do the validation.

In order to let the unit test verify the http response headers, the evhttp_add_header() should be called before invoking `s_injected_send_reply()`